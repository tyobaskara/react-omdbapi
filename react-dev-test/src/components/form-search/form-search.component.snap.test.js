import React from 'react';
import renderer from 'react-test-renderer';
import FormSearch from './form-search.component';

jest
  .mock('../../redux/movies/movies.selectors', () => ({
    selectSearchQuery: () => '',
    selectSearchOptions: () => [],
    selectMoviesPage: () => {},
    selectMoviesList: () => []
  }));

test('FormSearch snaptest', () => {
  const props = {
    searchQuery: '', 
    searchOptions: [],  
    moviesPage: {},
    moviesList: [],
    setSearchQuery: () => {},
    setSearchOptions: () => {},
    setMoviesList: () => {},
    setIsLoadingSearch: () => {},
    setIsSearchError: () => {},
    setMoviesPage: () => {}
  };

  const component = renderer.create(
    <FormSearch {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FormSearch render options snaptest', () => {
  const props = {
    searchQuery: '', 
    searchOptions: [{
      Title: 'Title',
      Year: 'Year',
      imdbID: 'imdbID'
    }],  
    moviesPage: {},
    moviesList: [],
    setSearchQuery: () => {},
    setSearchOptions: () => {},
    setMoviesList: () => {},
    setIsLoadingSearch: () => {},
    setIsSearchError: () => {},
    setMoviesPage: () => {}
  };

  const component = renderer.create(
    <FormSearch {...props} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});