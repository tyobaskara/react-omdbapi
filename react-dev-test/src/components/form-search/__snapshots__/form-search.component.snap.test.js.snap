// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch render options snaptest 1`] = `
<div
  className="form-search"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        autoComplete="off"
        id="searchMovies"
        name="searchMovies"
        onChange={[Function]}
        placeholder="Search Movies.."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </div>
  </form>
  <ul
    className="options"
  >
    <li
      onClick={[Function]}
    >
      Title
       - 
      Year
    </li>
  </ul>
</div>
`;

exports[`FormSearch snaptest 1`] = `
<div
  className="form-search"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        autoComplete="off"
        id="searchMovies"
        name="searchMovies"
        onChange={[Function]}
        placeholder="Search Movies.."
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        <i
          className="fa fa-search"
        />
      </button>
    </div>
  </form>
  <ul
    className="options"
  />
</div>
`;
